paimon-core/src/main/java/org/apache/paimon/KeyValueFileStore.java

```diff
@@ -93,7 +96,16 @@ public KeyValueFileStore(
         this.keyValueFieldsExtractor = keyValueFieldsExtractor;
         this.mfFactory = mfFactory;
         this.keyComparatorSupplier = new KeyComparatorSupplier(keyType);
-        this.valueEqualiserSupplier = new ValueEqualiserSupplier(valueType);
+        this.userDefinedSeqComparator = UserDefinedSeqComparator.create(valueType, options);
+        this.projection =
+                getProjectionWithIgnoreFields(
```

Comment:
I think you should judge `(changelogRowDeduplicate && !ignoreFields.isEmpty())` first before you creating ignore fields valueEqualiserSupplier

Reply:
I've judge it in getProjectionWithIgnoreFields method

Reply:
fixed